grace/evaluation/inference.py

```diff
@@ -1,184 +1,264 @@
 
-        # Iterate through each node, storing the variables:
-        for idx, node in G.nodes(data=True):
-            prediction = [int(n_pred[idx].item()), n_probabs[idx].numpy()]
+        # Iterate through each node, storing the 3 class predictions:
+        for n_idx, node in G.nodes(data=True):
+            prediction = Prediction(np.append(n_probs[n_idx], 0.0))
             node[GraphAttrs.NODE_PREDICTION] = prediction
 
         for e_idx, edge in enumerate(G.edges(data=True)):
-            prediction = (int(e_pred[e_idx].item()), e_probabs[e_idx].numpy())
+            prediction = Prediction(np.append(e_probs[e_idx], 0.0))
             edge[-1][GraphAttrs.EDGE_PREDICTION] = prediction
 
-    def visualise_model_performance_on_graph(
-        self, G: nx.Graph, positive_class: int = 1
-    ):
-        # Prep the data & plot them:
-        node_true = np.array(
-            [
-                node[GraphAttrs.NODE_GROUND_TRUTH]
-                for _, node in G.nodes(data=True)
-            ]
+    def visualise_prediction_probs_on_graph(self, G: nx.Graph) -> None:
+        """Visualise predictions as shaded points (nodes) / lines (edges)."""
+        assert all(GraphAttrs.NODE_PREDICTION in G.nodes[n] for n in G.nodes)
+        assert all(GraphAttrs.EDGE_PREDICTION in G.edges[e] for e in G.edges)
+        fig = visualise_node_and_edge_probabilities(G=G)
+        return fig
+
+    def get_predictions_for_entire_batch(
+        self,
+        infer_target_list: list[dict[str]],
+        verbose: bool = False,
+    ) -> tuple[npt.NDArray]:
+        """Processes predictions for list of graph targets."""
+
+        data = [[] for _ in range(6)]
+
+        for target in tqdm(infer_target_list, disable=not verbose):
+            # Read & update the graph:
+            G = target["graph"]
+            self.set_node_and_edge_probabilities(G=G)
+
+            # Store the node & edge data into vectors:
+            for _, n in G.nodes(data=True):
+                label = n[GraphAttrs.NODE_GROUND_TRUTH]
+                if label == Annotation.UNKNOWN:
+                    continue
+                data[0].append(label)
+                data[1].append(n[GraphAttrs.NODE_PREDICTION].label)
+                data[2].append(n[GraphAttrs.NODE_PREDICTION].prob_TP)
+
+            for _, _, e in G.edges(data=True):
+                label = e[GraphAttrs.EDGE_GROUND_TRUTH]
+                if label == Annotation.UNKNOWN:
+                    continue
+                data[3].append(label)
+                data[4].append(e[GraphAttrs.EDGE_PREDICTION].label)
+                data[5].append(e[GraphAttrs.EDGE_PREDICTION].prob_TP)
+
+        # Sanity checks:
+        assert len(data[0]) == len(data[1]) == len(data[2])
+        assert len(data[3]) == len(data[4]) == len(data[5])
+        data = [np.array(item) for item in data]
+
+        n_true, n_pred, n_prob, e_true, e_pred, e_prob = data
+        return n_true, n_pred, n_prob, e_true, e_pred, e_prob
```

Comment:
why not return data if in calculate_numerical_results the inverse operation is done?